### PR TITLE
Removes Construction Site

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ideally, this will be merged before: https://github.com/ParadiseSS13/Paradise/pull/13632 as that'll make things a lot easier for both of us to work around; also see further explanation below.

As it'll make the ruin refactor all that much more impactful and allow very rapid standardization of maps.

This removes the fourth z-level, known as "construction site".

In general, this place is almost never utilized for its actual intended function; building up a station.

Back in the day, when this was added, Paradise rounds lasted a lot longer and rounds didn't always end right at the 2 hour mark--this gave engineers a lot more time to actually build something.

Even so, it was very rarely utilized---once in a blue moon it was used to teach a new engineer a thing or two---orrrr, more often, it was used for antag shennanigans.

And that's its current status to this day; it's often used as a place to retreat to (if things go south) or an easy way to set up a cult base/do other antaggy things awfully easily.

And for this, we pay a very high price: A significant increase in the cost of atmos if someone so much as starts trying to do stuff on the site; it bloats up the active turfs by a huge margin. Making the station pre-filled only helps this issue a bit; you're still going to have engineers coming and going and bloating up the active turfs from deconstructing things, building out the pipenets, and altering atmos machinery.....and usually for only 1-2 people.

Even if this wasn't a problem, it being a vector for "go there, do antaggy things" isn't great, in my opinion, either---the crew should largely be on station and focused on doing things there and interacting with each other.

If/when this is merged in, this frees up yet another z-level for yet *even more space ruins*.

It also makes mapping standardization a *heckuva* lot easier for @AffectedArc07. There's no engineering shuttle he has to awkwardly cram into Delta and Metastation, making his job easier and better paving the way for allowing us to have map rotation.

Engineering site was a nice place---but it's since turned into science outpost 2.0; a place largely not used for its intended purpose---only this one is also a huge performance cost vector.

This also removes metastation z-4, in favor of an empty z-level; it's just the old derelict that we *used* to have.

And lastly, this cleans up the shuttle travel area for the science shuttle---which hasn't existed since 2017.

img

:cl: Fox McCloud
del: Removes construction site
/:cl:

### CODE REVIEW

**Comment by Balthusdire** [comment]
I'm all for this one. It was cool but it just isn't used and if it costs that much to the server then it isnt worth it.

**Comment by SteelSlayer** [comment]
Along with the construction site, we also had the engi outpost, which had some supplies, something like 5 stacks of metal and glass, and some wood as well. It would be nice if those were placed on-station somewhere, perhaps in that one crate in secure storage which already has metal/rods/glass and such.

**Comment by TDSSS** [comment]
Can you leave the construction site in as a map template that admins can place if desired? I've seen it used sometimes for things like building events and so on, and it couldn't hurt, right?

**Comment by Shockpoint** [comment]
I have never seen this as "science outpost 2.0" outside of the rare cult base that gets easily smashed to bits.

In any case, if we're going to remove this, we should have a place on station dedicated for new engineers/atmos techs to *learn* stuff without endangering the rest of the station. I always used the construction site if I was going to teach people, because it was safe and easy to manage, and any problems were confined to that place alone.

**Comment by Mitchs98** [comment]
I genuinely disagree with removing the engineering outpost. The construction site? Maybe, I can see the reasoning there for the whole atmos problems and all. But the engineering outpost I really don't often see used for antaggy things or anything, more often than not it's used to escape an onstation threat and/or decorated into a neat lil hideout area.

That said I have used it for reasons Shockpoint stated before, and testing things myself, so it would definitely be a good thing to add a similar area to our station z-level. Overall I don't see much of a reason to remove it myself other than to save a tiny bit of CPU(Or a lot, no differences were really added to compare on that front?).

**Comment by Rurik123** [comment]
While I completely agree the crew should mostly be on station and focused there, removing the engi outpost is a really bad idea. Its doesn't even come close to the levels of the science outpost, and camping engi outpost has its drawbacks unlike lavaland/gateway camping, because unlike lavaland/gateway it is easily checked and searched by sec and it is often done. I can't count how many interesting fights I've had there as both sec and antag. 

For that reason alone I think it should be kept, but there's also the supplies as the others mentioned, as well as a popular place to hang out or retreat to when there's too much chaos on station. Can't forget its a popular place to remodel and just test things as well. 

The outpost is a positive addition to the server as it has been as long as I've been here. Its presence makes things more interesting when it comes to the sec vs antag dynamic and I'm sure most sec and antag players would agree on that. If it were removed, our experiences would be lessened for it. 

I sincerely hope you change your opinion on this, as, aside from the reasons listed above, I'd miss it greatly.

There's also something to be said about how much cultists will suffer from this (more construction site than the engi outpost) but that's another tangent entirely.

**Comment by farie82** [comment]
Mostly echoing the stuff posted prior. But the engineering outpost was indeed mostly used by antags. But it was one of the only places antags could go to to be stealthy for a bit. All the while security could still reach them.

The station is rather crowded at the moment so being stealthy onboard is usually not an option. I also think it will lead to more antags just waiting out the round and then doing something at the end since they will have no place to hide after. Or they just go into space and security won't ever have a chance of finding them again till the end of the round.

Removing the construction site is... meh but sure if it's that expensive for the server. Having a template would be neat for admins to hold events with.

File: _maps/metastation.dm [hunk 1]
```diff
@@ -17,7 +17,7 @@ z7 = empty space
 	#include "map_files\MetaStation\MetaStation.v41A.II.dmm"
 	#include "map_files\MetaStation\z2.dmm"
 	#include "map_files\generic\tcommsat-blown.dmm"
-	#include "map_files\MetaStation\z4.dmm"
+	#include "map_files\generic\z4.dmm"
 	#include "map_files\generic\Lavaland.dmm"
 	#include "map_files\generic\z6.dmm"
 	#include "map_files\generic\z7.dmm"
```
Comment (AffectedArc07) [inline _maps/metastation.dm:23]: Meta's Z4 is the cyberiad Z6 from 4 years ago, not the engineering outpost.

This seems unintentional unless youre also killing that, which if so, I would add that to the PR description. 

Also if this map is no longer used, shouldn't it be removed from the files and `travis_map_testing.dm`

File: _maps/metastation.dm [hunk 2]
```diff
@@ -28,10 +28,10 @@ z7 = empty space
 DECLARE_LEVEL(MAIN_STATION, CROSSLINKED, list(STATION_LEVEL, STATION_CONTACT, REACHABLE, AI_OK)),\
 DECLARE_LEVEL(CENTCOMM, SELFLOOPING, list(ADMIN_LEVEL, BLOCK_TELEPORT, IMPEDES_MAGIC)),\
 DECLARE_LEVEL(TELECOMMS, CROSSLINKED, list(REACHABLE, BOOSTS_SIGNAL, AI_OK)),\
```
Comment (AffectedArc07) [inline _maps/metastation.dm:30]: Tcomms shouldnt have `BOOSTS_SIGNAL` or `AI_OK` anymore

File: code/modules/arcade/mob_hunt/mob_datums.dm [hunk 3]
```diff
@@ -139,7 +139,7 @@
 		for(var/areapath in typesof(A))
 			possible_areas[areapath] -= 2
 	//removes "bad areas" which shouldn't be on-station but are subtypes of station areas. probably should the unused ones and consider repathing the rest
-	var/list/bad_areas = list(subtypesof(/area/construction), /area/solar/derelict_starboard, /area/solar/derelict_aft, /area/solar/constructionsite)
```
Comment (AffectedArc07) [inline code/modules/arcade/mob_hunt/mob_datums.dm:142]: What

File: code/game/area/Space Station 13 areas.dm [hunk 4]
```diff
@@ -1867,76 +1867,6 @@ GLOBAL_LIST_EMPTY(ghostteleportlocs)
 
 //HALF-BUILT STATION (REPLACES DERELICT IN BAYCODE, ABOVE IS LEFT FOR DOWNSTREAM)
```
Comment (AffectedArc07) [inline code/game/area/Space Station 13 areas.dm:1868]: Remove this comment, it was never correct (we still had derelict), and the areas it references are gone.

**Comment by AffectedArc07** [comment]
A few non-code things I want to comment on:
- This still keeps the shuttle transit location on Z2, and considering how many of those we have, its probably worth clearing it out
- I see about 1 round a month where engineers actually use this to build on
- Filling this place with active turfs knocks the number up to **8000**, which will have the atmos SS  be paused every time it processes by the StonedMC 
- Its currently a hayday for rogue AIs because of the amount of APCs it has
- This doesn't affect the Ruin Rework in a huge way so doesnt hugely matter if it goes in before or after
- The outpost currently has a plethora of mapping issues, here they are in no particular order:

Invalid areas:
image

Pipes in walls:
image

These doors are so blatantly copypasted that the person who did them didnt even remove the wires
image

Why is this one singular tile grimy
image

These **wooden** floors are just regular floors with a different icon state
image


All in all, this needs a fixup or a removal, and considering the sheer beating this place puts on the atmos SS, I side with removal, because before long, people will discover the lag that this puts on the atmos SS, and it will become a DoS vector. I know that seems far-fetched, but never underestimate this community.

**Comment by Spacemanspark** [comment]
Regardless of what happens, I feel that removing an area that allows new players to semi-safely learn how to set up the singularity isn't the best idea. This isn't like the science outpost (may it rest in peace), which became functionally pointless once xenoarch was removed. It's remove made sense, regardless of how fun it was to siege upon during team antag rounds. 

I would rather see the outpost overall reworked than removed. I understand there are a few issues with it

As far as it being used as a cult base, it doesn't actually happen as often as some claim... and it's typically an area security checks first when cult is found anyways.
As far as APCs work, I really wish we only allowed AIs to convert on station APCs. Regardless, it doesn't start with cameras, and while an engineering cyborg can easily place some it's _incredibly_ risky, as any engineer that decides to go there will immediately spot the blue.

File: _maps/metastation.dm (hunk 1 above)
Comment (Fox-McCloud) [reply _maps/metastation.dm:23]: Oh, not intentional, but might as well.

File: code/modules/arcade/mob_hunt/mob_datums.dm (hunk 3 above)
Comment (Fox-McCloud) [reply code/modules/arcade/mob_hunt/mob_datums.dm:142]: Yeah, I know.

**Comment by procdrone** [comment]
If this is all about me making scrubber network and extending vent systems on the south side of the site, and pressurizing these areas im sorry I were just a bored engineer trying to do something.

@AffectedArc07 
>Its currently a hayday for rogue AIs because of the amount of APCs it has

And not a single camera so malf AI can't do nothing.

And to all you cult haven types. Ever since lavaland, construction site cult bases are more and more rare, giving in to superior choice of deep lavaland bases. 

If a cult goes for a space base, it will rather be something like whiteship, DJ station or other obscure locations.

If change must be made for performance sake, a workoever of the area would be much more preferable (IMO) than outright cutting all of it out.

**Comment by AffectedArc07** [comment]
> And not a single camera so malf AI can't do nothing.

The AI just sends a borg over to stand by APCs while they hack them since borgs have integrated cameras.

**Comment by procdrone** [comment]
> > And not a single camera so malf AI can't do nothing.
> 
> The AI just sends a borg over to stand by APCs while they hack them since borgs have integrated cameras.

Technically possible, but personally, never saw that happen myself though.

Astute crew could, theoretically notice the borg missing and check its inbuilt camera and notice AI up to no good. 

To fix this issue, could give a check for APCs so only these on the z level 1 can be hacked.

But that's very speculative at this point, as again, never saw that happen. And the PR not about this.

**Comment by Rurik123** [comment]
> > > And not a single camera so malf AI can't do nothing.
> > 
> > 
> > The AI just sends a borg over to stand by APCs while they hack them since borgs have integrated cameras.
> 
> Technically possible, but personally, never saw that happen myself though.

I've seen it happen once over my entire stay at paradise. So, I'd say a rare neat play of teamwork with the AI and borg shouldn't be a reason for removal.

> And to all you cult haven types. Ever since lavaland, construction site cult bases are more and more rare, giving in to superior choice of deep lavaland bases.
> 
> If a cult goes for a space base, it will rather be something like whiteship, DJ station or other obscure locations.

I feel I should draw attention to this as it is a good point. As sec, dealing with cult lavaland or cult on the white ship is infinitely more annoying, lavaland cult simply because its a giant fucking hassle to get to their culty spot and white ship because they can build a few constructs, then abuse the launching system to knockdown invading sec permanently (via transferring to a different z level over and over again) while the constructs kill them.

Construction cult, in my experience, is more common but not the better for it. Then again I have been out of the loop for the last month.

**Comment by AffectedArc07** [comment]
Ok, not fully related to this PR, but mapdiffbot is fucked. It says this is **Meta's** Z4:
image

When Meta's Z4 is this:
image

**Comment by Spacemanspark** [comment]
The outpost clinging to life

**Comment by Oyu-De** [comment]
> > > > And not a single camera so malf AI can't do nothing.
> 
> > > 
> 
> > > 
> 
> > > The AI just sends a borg over to stand by APCs while they hack them since borgs have integrated cameras.
> 
> > 
> 
> > Technically possible, but personally, never saw that happen myself though.
> 
> 
> 
> I've seen it happen once over my entire stay at paradise. So, I'd say a rare neat play of teamwork with the AI and borg shouldn't be a reason for removal.

1: Whenever I have a malf AI, first thing they do is this.
“Work with the crew for now, when they forget about you, go to construction site” or something similar.

2: As for crew, no one checks for engineering borg. As for RD, although they should, they never check for borgs etc.

Basically, this is how, as a borg, malf Aı rounds go. Borgs choose either janitor, engineering or security, engineering borg goes to construction site, AI hacks them. If a witness exists, security borg killd them. t that point, it is over as it is free realestate for borg factory.

**Comment by Kyep** [comment]
Temporarily removed my approval of this while a question about loss of engineering hardsuits is answered.

**Comment by Fox-McCloud** [comment]
Moved the hardsuits, metals, glass, and wood to engineering storage.

**Comment by Rurik123** [comment]
Any chance for the flares/donks to be moved there as well? Though I suppose the donks wouldn't make too much sense anymore being it doesn't have a microwave to be paired with.

**Comment by AffectedArc07** [comment]
TIL mapdiffbot breaks if you have 2 maps with the same name

**Comment by taukausanake** [comment]
Couldn't the construction site be used as a station goal? Construct a new station to be used in the NT fleet of stations and also/maybe make up for the multiple self-destructs of the Cyberiad? You would also get all the departments working on it as each section would be in charge of putting their area together as they want. It's taking people off-station sure, but they're all in one place too

**Pull request merged**